src/Microsoft.AspNet.OData.Shared/Query/DefaultSkipTokenHandler.cs

```diff
@@ -0,0 +1,387 @@
+                }
+                else
+                {
+                    compare = binder.CreateBinaryExpression(BinaryOperatorKind.GreaterThan, property, constant, true);
+                }
+
+                if (firstProperty)
+                {
+                    lastEquality = binder.CreateBinaryExpression(BinaryOperatorKind.Equal, property, constant, true);
+                    where = compare;
+                    firstProperty = false;
+                }
+                else
+                {
+                    Expression condition = Expression.AndAlso(lastEquality, compare);
+                    where = where == null ? condition : Expression.OrElse(where, condition);
+                    lastEquality = Expression.AndAlso(lastEquality, binder.CreateBinaryExpression(BinaryOperatorKind.Equal, property, constant, true));
+                }
+            }
+
+            Expression whereLambda = Expression.Lambda(where, param);
+            return ExpressionHelpers.Where(query, whereLambda, query.ElementType);
+        }
+
+        /// <summary>
+        /// Generates a dictionary with property name and property values specified in the skiptoken value.
+        /// </summary>
+        /// <param name="value">The skiptoken string value.</param>
+        /// <param name="context">The <see cref="ODataQueryContext"/> which contains the <see cref="IEdmModel"/> and some type information</param>
+        /// <returns>Dictionary with property name and property value in the skiptoken value.</returns>
+        protected static IDictionary<string, object> PopulatePropertyValuePairs(string value, ODataQueryContext context)
+        {
+            Contract.Assert(context != null);
+
+            IDictionary<string, object> propertyValuePairs = new Dictionary<string, object>();
+            string[] keyValues = value.Split(CommaDelimiter);
+            foreach (string keyAndValue in keyValues)
+            {
+                string[] pieces = keyAndValue.Split(new char[] { propertyDelimiter }, 2);
+                if (pieces.Length > 1 && !String.IsNullOrWhiteSpace(pieces[1]))
+                {
+                    object propValue = null;
+                    if (pieces[1].StartsWith("'enumType'"))
+                    {
+                        string enumValue = pieces[1].Remove(0, 10);
+                        IEdmTypeReference type = EdmLibHelpers.GetTypeReferenceOfProperty(context.Model, context.ElementClrType, pieces[0]);
+                        propValue = ODataUriUtils.ConvertFromUriLiteral(enumValue, ODataVersion.V401, context.Model, type);
+                    }
+                    else
+                    {
+                        propValue = ODataUriUtils.ConvertFromUriLiteral(pieces[1], ODataVersion.V401);
```

Comment:
>ODataVersion.V401 [](start = 83, length = 17)

Is there any reason to use 4.01 version?

Reply:
We can use any of the versions and this is independent of the version for the service. Therefore I decided to use the latest. 

---
In reply to: [282591750](https://github.com/OData/WebApi/pull/1724#discussion_r282591750) [](ancestors = 282591750)